src/sql/base/parts/tree/browser/treeModel.ts

```diff
@@ -14,21 +14,23 @@ interface IMap<T> { [id: string]: T; }
 interface IItemMap extends IMap<Item> { }
 interface ITraitMap extends IMap<IItemMap> { }
 
-export class LockData {
+export class LockData extends Disposable {
 
 	private _item: Item;
-	private _onDispose?= new Emitter<void>();
+	private _onDispose?= this._register(new Emitter<void>());
 	readonly onDispose: Event<void> = this._onDispose!.event;
 
 	constructor(item: Item) {
+		super();
 		this._item = item;
 	}
 
 	get item(): Item {
 		return this._item;
 	}
 
-	dispose(): void {
+	override dispose(): void {
+		super.dispose();
 		if (this._onDispose) {
```

Comment:
You're going to be disposing this before calling it. I'd suggest looking into how this is used to understand how best to dispose the emitter (it's possible what it had before is correct)